depeche_db/_aggregated_stream.py

```diff
@@ -547,52 +562,137 @@ def update_full(self, budget: Optional[TimeBudget] = None) -> int:
+            _sa.select(
+                message_table.c.stream,
+                _sa.func.max(message_table.c.version).label(
+                    "max_origin_stream_version"
+                ),
+                _sa.func.min(message_table.c.global_position).label(
+                    "min_global_position"
+                ),
+            )
+            .where(
+                _sa.and_(
+                    _sa.or_(
+                        *[
+                            message_table.c.stream.like(wildcard)
+                            for wildcard in self.stream_wildcards
+                        ]
+                    ),
+                    *cutoff_cond,
+                )
+            )
+            .group_by(message_table.c.stream)
+            .cte("max_origin_stream_version")
+        )
+        max_aggregated_stream_version = (
             _sa.select(
                 tbl.c.origin_stream,
-                _sa.func.max(tbl.c.origin_stream_version).label("max_version"),
+                _sa.func.max(tbl.c.origin_stream_version).label(
+                    "max_aggregated_stream_version"
+                ),
             )
             .group_by(tbl.c.origin_stream)
-            .cte()
+            .cte("max_aggregated_stream_version")
+        )
+        streams_to_be_updated = (
+            _sa.select(
+                max_origin_stream_version.c.stream,
+                _sa.func.coalesce(
+                    max_aggregated_stream_version.c.max_aggregated_stream_version, 0
+                ).label("max_aggregated_stream_version"),
+                max_origin_stream_version.c.min_global_position,
+                (
+                    max_origin_stream_version.c.max_origin_stream_version
+                    - _sa.func.coalesce(
+                        max_aggregated_stream_version.c.max_aggregated_stream_version, 0
+                    )
+                ).label("message_count"),
+            )
+            .select_from(
+                max_origin_stream_version.join(
+                    max_aggregated_stream_version,
+                    max_origin_stream_version.c.stream
+                    == max_aggregated_stream_version.c.origin_stream,
+                    isouter=True,
+                )
+            )
+            .where(
+                _sa.or_(
+                    max_aggregated_stream_version.c.max_aggregated_stream_version.is_(
+                        None
+                    ),
+                    max_origin_stream_version.c.max_origin_stream_version
+                    > max_aggregated_stream_version.c.max_aggregated_stream_version,
+                )
+            )
+            # Oldest (= first message the oldest) streams first
+            .order_by(max_origin_stream_version.c.min_global_position)
         )
+        selected_streams = []
+        message_count = 0
+        for row in conn.execute(streams_to_be_updated):
+            if message_count >= self.batch_size:
+                break
```

Comment:
Hmm, this will increase the likelihood of messages being NOT ordered according to their global position within their respective partitions. This order is not guaranteed but I tried to stick to it in the usual cases.

A way to keep the order would be to select (up to) `batch_size` streams. Then we would get closer to the "right" order.